src/lib/hooks/useNotificationHandler.ts

```diff
@@ -46,8 +46,9 @@ const DEFAULT_HANDLER_OPTIONS = {
   shouldSetBadge: true,
 }
 
-// This needs to stay outside the hook to persist between account switches
+// These need to stay outside the hook to persist between account switches
 let storedPayload: NotificationPayload | undefined
+let prevDate = 0
```

Comment:
Let's see if this helps with the notification issue, and if it does I'm going to look into what a more accurate fix is. Went over the docs again for expo-notifications and we're implementing the handler right, so not quite certain what is going on here. This should work though inbetween account switches.